### DISCUSSION TITLE
Can I link the desktop client to scli?

### DISCUSSION THREAD

**Discussion by feklee** [isamert/scli#135] [Q&A]
I would like to use scli on my PinePhone, and the desktop client on my PC (primarily for voice and video calls).

**Comment by exquo** [comment]
You just need a 'master' signal account. The typical way to go is to _register_ a signal account through a phone app (android or ios), and then _link_ it with signal-desktop and signal-cli. But you can also register a new account with signal-cli (which will then be used by scli) and link it with signal-desktop. Here's some info in signal-cli's wiki on [linking signal-desktop with signal-cli master](https://github.com/AsamK/signal-cli/wiki/Quickstart#use-signal-cli-to-register-an-account-without-a-phone-and-link-with-signal-desktop) without using android / ios devices. 

But I'd be curios to know how usable scli would be on a device without a physical keyboard. If it's convenient to use on touchscreen mobile devices, that would be great!

**Reply by feklee** [reply]
Thank you! I’m looking for a Signal client for use on my PinePhone with SXMO/postmarketOS. I don’t have an Android or iOS device.

**Reply by exquo** [reply]
There are a few additional alternatives: run [signal-android in anbox](https://reddit.com/r/pinephone/comments/mqybop/signal_on_pinephone_success/); [compile signal-desktop for pinephone](https://reddit.com/r/pinephone/comments/lcb5b8/compiling_signaldesktop_for_pinephone/); use a third-party mobile linux app like [axolotl](https://github.com/nanu-c/axolotl) or [whisperfish](https://gitlab.com/whisperfish/whisperfish); use a signal [bridge](https://github.com/exquo/signal-soft/wiki/Software-list#bridges) for matrix / XMPP / pidgin; and more - I've attempted to collect this info in a wiki here: https://github.com/exquo/signal-soft/wiki/Software-list. See also discussions in the [links section](https://github.com/exquo/signal-soft/wiki/Links) there.

If you do get scli working satisfactorily with a touchscreen though - let us know! I haven't yet heard of people doing it, and I haven't tried it myself.

**Reply by feklee** [reply]
Thanks for the collection! signal-android in anbox and signal-desktop are likely too resource hungy. I generally even avoid starting up Firefox or Chromium. The PinePhone is very weak. So light weight applications are needed. A matrix-signal bridge breaks E2E encryption.

**Reply by exquo** [reply]
> signal-android in anbox and signal-desktop are likely too resource hungy

Yes, probably more than even a full browser.

> A matrix-signal bridge breaks E2E encryption.

You can run a matrix server on the device, then the decryption happens there. Not very lightweight either though.
